src/d3a/d3a_core/util.py

```diff
@@ -503,3 +503,8 @@ def area_sells_to_child(trade, area_name, child_names):
 def child_buys_from_area(trade, area_name, child_names):
     return area_name_from_area_or_iaa_name(trade.buyer) == \
         area_name and area_name_from_area_or_iaa_name(trade.seller) in child_names
+
+
+def if_not_in_list_append(target_list, obj):
+    if obj not in target_list:
+        target_list.append(obj)
```

Comment:
Better move such utility functions to d3a-interface (optional for now though)

Reply:
@faizan2590 If you don mind and in the sake of time i would leave it here for now, ok?